### PR TITLE
Alerting: Add support for Alert State History Loki primary

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

This PR adds support for showing the new Alert State History visualization when multiple backends have been configured with Loki as the primary.

**Special notes for your reviewer:**

Tested with
```
[unified_alerting.state_history]
enabled = true
backend = multiple
primary = loki
secondaries = annotations
loki_remote_url = http://localhost:3100

[feature_toggles]
enable = alertStateHistoryLokiPrimary, alertStateHistoryLokiSecondary
```

### CODE REVIEW

File: packages/grafana-data/src/types/config.ts [hunk 1]
```diff
@@ -64,6 +64,8 @@ export interface UnifiedAlertingConfig {
   minInterval: string;
   // will be undefined if alerStateHistory is not enabled
   alertStateHistoryBackend?: string;
+  // will be undefined if implementation is not "multiple"
```
Comment (gillesdemey) [inline packages/grafana-data/src/types/config.ts:67]: @alexweav can I bother you to double check this assumption? Testing it locally I was only getting the primary for the `multiple` backend

File: pkg/api/frontendsettings.go [hunk 2]
```diff
@@ -236,6 +236,7 @@ func (hs *HTTPServer) getFrontendSettings(c *contextmodel.ReqContext) (*dtos.Fro
 
 	if hs.Cfg.UnifiedAlerting.StateHistory.Enabled {
 		frontendSettings.UnifiedAlerting.AlertStateHistoryBackend = hs.Cfg.UnifiedAlerting.StateHistory.Backend
+		frontendSettings.UnifiedAlerting.AlertStateHistoryPrimary = hs.Cfg.UnifiedAlerting.StateHistory.MultiPrimary
```
Comment (gillesdemey) [inline pkg/api/frontendsettings.go:239]: The variable is called `MultiPrimary` which I'm assuming just means `primary` when `multiple` backend is configured and not that there could be multiple primaries?

Comment (gotjosh) [reply pkg/api/frontendsettings.go:239]: That's correct.

File: public/app/features/alerting/unified/hooks/useStateHistoryModal.tsx [hunk 3]
```diff
@@ -20,10 +20,18 @@ function useStateHistoryModal() {
 
   const styles = useStyles2(getStyles);
 
-  const implementation =
```
Comment (gotjosh) [inline public/app/features/alerting/unified/hooks/useStateHistoryModal.tsx:23]: Do we need a test for this?

**Review by gotjosh — APPROVED** [review]
LGTM

**Review by gotjosh — APPROVED** [review]
LGTM

File: packages/grafana-data/src/types/config.ts (hunk 1 above)
Comment (gotjosh) [reply packages/grafana-data/src/types/config.ts:67]: This is correct as well.

**Pull request merged**